### PR TITLE
*: upgrade golang version to 1.20

### LINKED ISSUE #8327: upgrade to 1.20
## Enhancement

The latest Go release, version 1.20, arrives six months after [Go 1.19](https://tip.golang.org/doc/go1.19). Most of its changes are in the implementation of the toolchain, runtime, and libraries. As always, the release maintains the Go 1 [promise of compatibility](https://tip.golang.org/doc/go1compat).

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #8327 #8320

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by D3Hunter** [comment]
CI golang version will be updated at friday. some test will fail before that

File: deployments/ticdc/docker/integration-test.Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ RUN ./download-integration-test-binaries.sh master
 RUN ls ./bin
 
 # Download go into /usr/local dir.
-ENV GOLANG_VERSION 1.19
+ENV GOLANG_VERSION 1.20
 ENV GOLANG_DOWNLOAD_URL https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz
 ENV GOLANG_DOWNLOAD_SHA256 464b6b66591f6cf055bc5df90a9750bf5fbc9d038722bb84a9d56a2bea974be6
```
Comment (hi-rustin) [inline deployments/ticdc/docker/integration-test.Dockerfile:16]: Please update sha.

**Review by hi-rustin — CHANGES_REQUESTED** [review]

File: deployments/ticdc/docker/integration-test.Dockerfile (hunk 1 above)
Comment (overvenus) [reply deployments/ticdc/docker/integration-test.Dockerfile]: @hi-rustin Could you add a CI to test docker build? I think it's very easy to forget to update hash cause people usually do a  batch search and replace.

**Review by lance6716 — APPROVED** [review]

File: deployments/ticdc/docker/integration-test.Dockerfile (hunk 1 above)
Comment (hi-rustin) [reply deployments/ticdc/docker/integration-test.Dockerfile]: We do not have docker in docker test in our Jenkin CI. So it's difficult to test it. Maybe we can add a comment here for now.

Comment (hi-rustin) [reply deployments/ticdc/docker/integration-test.Dockerfile]: CC: @D3Hunter

Comment (hi-rustin) [reply deployments/ticdc/docker/integration-test.Dockerfile]: We should let people notice that we need to change the sha here.

Comment (D3Hunter) [reply deployments/ticdc/docker/integration-test.Dockerfile]: don't think there's need to verify sha for test, better remove it

**Review by wuhuizuo — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**